src/hooks/blogs.ts

```diff
@@ -0,0 +1,40 @@
+import { useEffect } from "react";
+import { useDispatch, useSelector } from "react-redux";
+import api from "../api";
+import * as blogSelector from "../state/blog/selector";
+import * as blogActions from "../state/blog/actions";
+
+const getBlogPosts = async (fetch: string) => {
+    try {
+        let response
+        if(fetch === 'latest') {
+            response = await api.blogs.getLatestBlogs()
+        } else {
+            response = await api.blogs.getBlogPosts()
+        }
+        
+        return response.data.posts;
+    } catch(error) {
+        if (process.env.NODE_ENV === 'development') {
+            console.log(error)
+        }
+        throw error;
+    }
+}
+
+const usePosts = (fetch='all') => {
+    const dispatch = useDispatch();
+    const posts = useSelector(blogSelector.posts)
+    
+    useEffect(() => {
+        (async () => {
+            dispatch(blogActions.setPosts(await getBlogPosts(fetch)))
+        })();
+    }, [dispatch])
```

Comment:
Currently, the code is compiling with the warning that the dependency array is missing `fetch`. To fix this, either include it in the dependency array, `[dispatch, fetch]`, or ignore the warning by placing `// eslint-disable-next-line` on the line above. (easier to review actual errors if there are no warnings)

Reply:
resolved